core/invoker/src/main/scala/whisk/core/invoker/InvokerReactive.scala

```diff
@@ -130,30 +130,41 @@ class InvokerReactive(
             s"posted ${if (recovery) "recovery" else "completion"} of activation ${activationResult.activationId}")
       }
     }
-    // Potentially sends activation metadata to kafka if user events are enabled
-    UserEvents.send(
-      producer, {
+
+    if (UserEvents.enabled) {
```

Comment:
Minor nit. Currently [coverage for this code path is nil][1]. May be we refactor this to a utility object and test the code here. Or enable user events for test setup

[1]: https://codecov.io/gh/apache/incubator-openwhisk/pull/4006/diff#D1-134

Reply:
Now this should get proper coverage!